.github/workflows/doxygen.yml

```diff
@@ -0,0 +1,30 @@
+name: Doxygen
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  Build and Publish Documentation:
+    name: Build and Publish Documentation
+    runs-on: ubuntu-latest
+    container:
+      image: novelrt/novelrt-build:latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Generate Documentation
+        run: scripts/cibuild.sh --documentation -DDOXYGEN_OUTPUT_DIR=$GITHUB_WORKSPACE/novelrt/docs
+
+      - name: Commit Documentation Updates
+        if: ${{ success() }}
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git checkout gh-pages
+          git add ./docs/*
+          git commit -m "Updated documentation - $GITHUB_SHA"
```

Comment:
Updated to use the SHA of the current commit on master (that triggers this workflow) if I've understood the docs correctly.